### ISSUE TITLE
Error building SAI/meta makefile

### ISSUE THREAD

**Issue by gck-04** [Azure/sonic-sairedis#772]
Hi,
I am trying to execute Makefile in SAI/meta/ directory. But the build is failing. Please find the log below

----@ubuntu:~/sonic_activity/sonic-sairedis$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
-----@ubuntu:~/sonic_activity/sonic-sairedis$ uname -a\
> ^C
------@ubuntu:~/sonic_activity/sonic-sairedis$ uname -a
Linux ubuntu 5.8.0-38-generic #43~20.04.1-Ubuntu SMP Tue Jan 12 16:39:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
vtl-sonic@ubuntu:~/sonic_activity/sonic-sairedis$ make
make  all-recursive
make[1]: Entering directory '/home/vtl-sonic/sonic_activity/sonic-sairedis'
Making all in meta
make[2]: Entering directory '/home/vtl-sonic/sonic_activity/sonic-sairedis/meta'
make -C ../SAI/meta
make[3]: Entering directory '/home/vtl-sonic/sonic_activity/sonic-sairedis/SAI/meta'
perl -I. parse.pl
Processing capability file sample.cap
Skipping cap file sample.cap
Running Aspell
Words to check: 2572
Processing sai_8h.xml
Removing last element SAI_API_MAX
Processing saiacl_8h.xml
WARNING: attribute SAI_ACL_TABLE_ATTR_USER_DEFINED_FIELD_GROUP_MIN must have @range tag
WARNING: attribute SAI_ACL_ENTRY_ATTR_USER_DEFINED_FIELD_GROUP_MIN must have @range tag
Processing saibfd_8h.xml
WARNING: type 'const sai_bfd_session_state_notification_t *' is pointer, @count is required, but missing on sai_bfd_session_state_change_notification_fn
Processing saibridge_8h.xml
Processing saibuffer_8h.xml
Processing saicounter_8h.xml
Processing saidebugcounter_8h.xml
Processing saidtel_8h.xml
Removing last element SAI_DTEL_EVENT_TYPE_MAX
Processing saiexperimentalbmtor_8h.xml
Processing saiextensions_8h.xml
Processing saifdb_8h.xml
WARNING: type 'const sai_fdb_event_notification_data_t *' is pointer, @count is required, but missing on sai_fdb_event_notification_fn
Processing saihash_8h.xml
Processing saihostif_8h.xml
WARNING: type 'const void *' is pointer, @count is required, but missing on sai_packet_event_notification_fn
Use of uninitialized value $countParamName in pattern match (m//) at parse.pl
	line 880 (#1)
    (W uninitialized) An undefined value was used as if it were already
    defined.  It was interpreted as a "" or a 0, but maybe it was a mistake.
    To suppress this warning assign a defined value to your variables.
    
    To help you figure out what was undefined, perl will try to tell you
    the name of the variable (if any) that was undefined.  In some cases
    it cannot do this, so it also tells you what operation you used the
    undefined value in.  Note, however, that perl optimizes your program
    and the operation displayed in the warning may not necessarily appear
    literally in your program.  For example, "that $foo" is usually
    optimized into "that " . $foo, and the warning will refer to the
    concatenation (.) operator, even though there is no . in
    your program.
    
Use of uninitialized value $countParamName in hash element at parse.pl line 886 (#1)
Use of uninitialized value $countParamName in concatenation (.) or string at
	parse.pl line 888 (#1)
WARNING: count param name '' on sai_packet_event_notification_fn is not defined
Processing saiipmc_8h.xml
Processing saiipmcgroup_8h.xml
Processing saiisolationgroup_8h.xml
Processing sail2mc_8h.xml
Processing sail2mcgroup_8h.xml
Processing sailag_8h.xml
Processing saimacsec_8h.xml
Processing saimcastfdb_8h.xml
Processing saimetadatalogger_8h.xml
Processing saimetadatatypes_8h.xml
Processing saimetadatautils_8h.xml
Processing saimirror_8h.xml
Processing saimpls_8h.xml
Processing sainat_8h.xml
Processing saineighbor_8h.xml
Processing sainexthop_8h.xml
Processing sainexthopgroup_8h.xml
Processing saiobject_8h.xml
Processing saipolicer_8h.xml
Processing saiport_8h.xml
Removing last element SAI_PORT_BREAKOUT_MODE_TYPE_MAX
Removing last element SAI_PORT_INTERFACE_TYPE_MAX
WARNING: type 'const sai_port_oper_status_notification_t *' is pointer, @count is required, but missing on sai_port_state_change_notification_fn
Processing saiqosmap_8h.xml
Processing saiqueue_8h.xml
WARNING: type 'const sai_queue_deadlock_notification_data_t *' is pointer, @count is required, but missing on sai_queue_pfc_deadlock_notification_fn
Processing sairoute_8h.xml
Processing sairouterinterface_8h.xml
Processing sairpfgroup_8h.xml
Processing saisamplepacket_8h.xml
Processing saischeduler_8h.xml
Processing saischedulergroup_8h.xml
Processing saisegmentroute_8h.xml
Processing saiserialize_8h.xml
Processing saistatus_8h.xml
Processing saistp_8h.xml
Processing saiswitch_8h.xml
Processing saiswitchextensions_8h.xml
Processing saisystemport_8h.xml
Processing saitam_8h.xml
WARNING: type 'const void *' is pointer, @count is required, but missing on sai_tam_event_notification_fn
WARNING: count param name '' on sai_tam_event_notification_fn is not defined
Processing saitunnel_8h.xml
Processing saitypes_8h.xml
Removing last element SAI_COMMON_API_MAX
Removing last element SAI_OBJECT_TYPE_MAX
Removing last element SAI_PORT_ERR_STATUS_MAX
Processing saitypesextensions_8h.xml
Processing saiudf_8h.xml
Processing saivirtualrouter_8h.xml
Processing saivlan_8h.xml
Processing saiwred_8h.xml
ERROR: invalid type '' for SAI_ACL_COUNTER_ATTR_TABLE_ID
ERROR: flags are not defined for SAI_ACL_COUNTER_ATTR_TABLE_ID
ERROR: FATAL ERROR === MUST FIX === : Can't use an undefined value as an ARRAY reference at parse.pl line 1435.

make[3]: *** [Makefile:93: saimetadata.h] Error 1
make[3]: Leaving directory '/home/vtl-sonic/sonic_activity/sonic-sairedis/SAI/meta'
make[2]: *** [Makefile:1333: saimeta-gen] Error 2
make[2]: Leaving directory '/home/vtl-sonic/sonic_activity/sonic-sairedis/meta'
make[1]: *** [Makefile:412: all-recursive] Error 1
make[1]: Leaving directory '/home/vtl-sonic/sonic_activity/sonic-sairedis'
make: *** [Makefile:342: all] Error 2
-------@ubuntu:~/sonic_activity/sonic-sairedis$

**Comment by gck-04** [comment]
Hi @kcudnik could you please add your thoughts on this issue.

**Comment by kcudnik** [comment]
do you have PR  for this change ?

**Comment by kcudnik** [comment]
which branch you are trying to build?

**Comment by gck-04** [comment]
> which branch you are trying to build?

Hi Kamil, first of all I am sorry for the late reply to your previous question and thank you for being available whenever I needed assistance.

Come back to this issue I created, it was because of the doxygen version mismatch. Later I downloaded correct version of doxygen package and the issue got resolved.
That was the master branch with latest commit of that day.
